core/src/test/java/PlayerTests.java

```diff
@@ -0,0 +1,144 @@
+import org.mockito.internal.util.reflection.Whitebox;
+
+@RunWith(GdxTestRunner.class)
+public class PlayerTests {
+    TmxMapLoader mapLoader = new TmxMapLoader();
+    TiledMap map = mapLoader.load("assets/AuberMap.tmx");
+    MapGraph mapGraph = new MapGraph();
+    Node node = new Node(2416, 2768);
+    Node node2 = new Node(2446, 3312);
+    Sprite sprite = new Sprite(new Texture("assets/Tutorial3.png")); // for janky reasons
+    Player player;
+    Array<TiledMapTileLayer> playerCollisionLayers = new Array<>();
+
+    Infiltrator frozen_infiltrator;
+    CrewMembers not_frozen_crew;
+
+    /**
+     * Tests that players can take damage
+     */
+    @Test
+    public void TakeDamageTest() {
+        playerCollisionLayers.add((TiledMapTileLayer) map.getLayers().get("Tile Layer 1"));
+        playerCollisionLayers.add((TiledMapTileLayer) map.getLayers().get(2));
+        player = new Player(sprite, playerCollisionLayers, true);
+        player.takeDamage(30);
+        assertEquals("Error: Player can't take damage'", 70, player.getHealth(), 0.000001);
+    }
+
+    /**
+     * Tests that player can heal
+     */
+    @Test
+    public void HealTest() {
+        playerCollisionLayers.add((TiledMapTileLayer) map.getLayers().get("Tile Layer 1"));
+        playerCollisionLayers.add((TiledMapTileLayer) map.getLayers().get(2));
+        player = new Player(sprite, playerCollisionLayers, true);
+        player.takeDamage(30);
+        player.heal(20);
+        assertEquals("Error: Player can't heal'", 90, player.getHealth(), 0.000001);
+    }
+
+    /**
+     * Tests that the powerup ArrestUp increases the arrestable range
+     */
+    @Test
+    public void ArrestUpTest() {
+        playerCollisionLayers.add((TiledMapTileLayer) map.getLayers().get("Tile Layer 1"));
+        playerCollisionLayers.add((TiledMapTileLayer) map.getLayers().get(2));
+        player = new Player(sprite, playerCollisionLayers, true);
+        player.setPosition(Config.POWERUP_START_X, Config.POWERUP_START_Y);
+        ArrestUp arrestUp = new ArrestUp(new Vector2(Config.POWERUP_START_X, Config.POWERUP_START_Y));
+        arrestUp.update(player);
+        player.update(0.01f);
+        assertEquals("Error: ArrestUp powerup doesn't increase arrest range", Config.EXTEND_ARREST_RANGE,
+                ((Float) Whitebox.getInternalState(player, "arrestRadius")), 0.1f);
+    }
+
+    /**
+     * Tests that the powerup FreezeUp freezes infiltrators but not innocent crew members
+     */
+    @Test
+    public void FreezeUpTest() {
+        playerCollisionLayers.add((TiledMapTileLayer) map.getLayers().get("Tile Layer 1"));
+        playerCollisionLayers.add((TiledMapTileLayer) map.getLayers().get(2));
+        player = new Player(sprite, playerCollisionLayers, true);
+        mapGraph.addNode(node);
+        mapGraph.addNode(node2);
+        mapGraph.connectNodes(node, node2);
+
+        frozen_infiltrator = new Infiltrator(sprite, node, mapGraph);
```

Comment:
This is causing a null pointer exception

Reply:
> fixed!